### ISSUE TITLE
Markers vs Decorations: Markers do not track editing changes

### ISSUE THREAD

**Issue by bcdev-com** [microsoft/monaco-editor#938]
monaco-editor version: 0.13.1
Browser: Edge & Chrome (presumably all)
OS: Windows 10
Steps or JS usage snippet reproducing the issue:

1) add markers via monaco.editor.setModelMarkers
2) edit the document inserting or removing lines above the marker or characters before the marker
3) visually, the squiggly underlines remain with the text they were applied to, including expanding and contracting if typing in the middle of the marker, per TrackedRangeStickiness.NeverGrowsWhenTypingAtEdges
4) Monaco.editor.getModelMarkers({}) returns the markers with the original, no longer correct, start/end-Line/Column values.

So - it appears the decorations used to display the markers are tracking the model, but the markers themselves do not?  This seems very counter-intuitive.  If this is by design, do we lose anything just using Decorations directly instead of Markers?

Thanks,
-Blake

**Comment by alexandrudima** [comment]
To fully explain the differences between markers and decorations, I think it is best to use VS Code as an example.

#### Markers can exist without having a file opened

e.g. in this screenshot there are no models instantiated, yet the markers are still stored

![image](https://user-images.githubusercontent.com/13972192/35488184-69bef25e-0452-11e8-9e3b-b3a49f553336.png)

#### Markers get bound to a model when the URIs match

When the file is opened, the markers are converted into decorations and bound to the model. But, markers are never modified by the decorations, only the owner of the markers gets to set / modify them.

**Comment by szuni** [comment]
I tried markers with different playground samples.
It works as expected with https://microsoft.github.io/monaco-editor/playground.html#interacting-with-the-editor-line-and-inline-decorations
Marker moves to line 3 when I hit enter in first line.

Using https://microsoft.github.io/monaco-editor/playground.html#extending-language-services-configure-json-defaults when I hit enter marker moves to line 3 but few monents later it returns to line 2 but the marked content is not there anymore.


I used this code:
<pre>
monaco.editor.setModelMarkers(editor.getModel(), null, [
    {
            startLineNumber: 2,
            startColumn: 1,
            endLineNumber: 1,
            endColumn: 8,
            message: 'message',
            severity: 4
        }
]);
</pre>

How can I make it follow line feeds when I use JSON schema validation as well?
